### ISSUE TITLE
Building sys-apps/sandbox with -flto disables sandbox functionality

### ISSUE THREAD

**Issue by Peter-Levine** [InBetweenNames/gentooLTO#347]
I came across this while trying to figure out why a testcase that was supposed to fail due to a problem with sandbox was succeeding.  sys-apps/sandbox will build with `-flto` without issue.  But it will not adequately function to prevent sandbox access violations or report them.  This can be tested by starting a sandbox shell and trying to create a file outside of it, eg.:

```
cd /usr/local
sandbox
touch ~/testfile
```
When built without `-flto` flags, you get `Permission denied`.  But if built with `-flto`, the command succeeds without error.

**Comment by jiblime** [comment]
Are you sure you are running the commands correctly?
`touch ~/testfile` will create a testfile within your current user's home directory, which you (in a regular scenario) have rw access.
```
$ cd /usr/local
$ sandbox
$ touch testfile
touch: cannot touch 'testfile': Permission denied
$ touch ~/testfile
$ ls $_
/home/non-root-user/testfile
```

**Comment by javashin** [comment]
this seems to be happen to me too : 


sudo su
igloo-l440 /home/javashin/Desktop # cd /usr/local
igloo-l440 /usr/local # sandbox
============================= Gentoo path sandbox ==============================
Detection of the support files.
Verification of the required files.
Setting up the required environment variables.
The protected environment has been started.
--------------------------------------------------------------------------------
Process being started in forked instance.
igloo-l440 /usr/local # touch ~/testfile
igloo-l440 /usr/local # touch testfile
igloo-l440 /usr/local # 

see no Permission denied anywhere

**Comment by javashin** [comment]
with no lto : 


cd /usr/local/
igloo-l440 /usr/local # sandbox
============================= Gentoo path sandbox ==============================
Detection of the support files.
Verification of the required files.
Setting up the required environment variables.
The protected environment has been started.
--------------------------------------------------------------------------------
Process being started in forked instance.
igloo-l440 /usr/local # touch testfile ---- > creates fine testfile in /usr/local
igloo-l440 /usr/local # touch ~/.testfile
touch: cannot touch '/root/.testfile': Permission denied

**Comment by Peter-Levine** [comment]
> Are you sure you are running the commands correctly?
> `touch ~/testfile` will create a testfile within your current user's home directory, which you (in a regular scenario) have rw access.

Yes.  sandbox should prevent that action regardless of permission.  The example I gave was to demonstrate the issue for any user.  But even as root, something trivial like this should suffice:

```
cd /usr/local
sandbox
touch /testfile
```
With a functioning sandbox, you get:
```
 * ACCESS DENIED:  open_wr:      /testfile
 * ACCESS DENIED:  utimensat:    /testfile
touch: cannot touch '/testfile': Permission denied
```
But when sandbox is built with lto, the command succeeds and the file is created.

**Comment by ionenwks** [comment]
Tried as well (tried both built with gcc 8.3 and gcc 9.1 on a system where nothing else is built with lto), I get the same results. I'm amazed this issue exist to be honest, it's a pretty bad deal.

**Comment by jelinekto** [comment]
I can reproduce with every version currently available in the Portage tree.

**Comment by InBetweenNames** [comment]
Confirmed!

**Comment by InBetweenNames** [comment]
Wow, that's a pretty serious one.
